account-analytics/src/main/java/com.bobocode/AccountAnalytics.java

```diff
@@ -123,7 +147,11 @@ public BigDecimal calculateTotalBalance() {
      * @return a map where key is a letter and value is its count in all first names
      */
     public Map<Character, Long> getCharacterFrequencyInFirstNames() {
-        throw new UnsupportedOperationException("It's your job to implement this method"); // todo
+        return accounts.stream()
+                .map(Account::getFirstName)
+                .flatMapToInt(String::chars)
```

Comment:
Sure. Method `String#chars()` creates a `IntStream` of characters. (It's because `char` is basically an `int`, and JDK **doesn't provide a special stream class to hold primitive characters**). So, if you want to get a stream of primitive `char`, you get `IntStream`.
 
`accounts.stream().map(Account::getFirstName)` will return a `Stream<String>`, so if you will map each string  element into a stream of its chars, you will get `Stream<IntStream>` - **a stream of streams**.

`flatMapToInt(String::chars)` is a complex operation that consists of two basic operations:
- *"MapToInt"* which receives a mapper - function that maps `String` into an `IntStream`
- *"flat"* transforms `Stream<IntStream>` into a `IntStream`, e.g. ((3, 4), (3, 5, 7), (4)) -> (3, 4, 3, 5, 7, 4). **Makes it flat.**

Now, when you have an `IntStream` that consists of character codes, you want to transform it into a `Stream<Character>`, because after you will want to group them into a `Map<Character, Long>`

- `mapToObj(c -> (char) c)` uses a mapper that cast `int` to `char`, and then map `char` to object `Character`

Now, when you have a `Stream<Character>`, you can collect them into a map. 

- `groupingBy(Function.identity(), counting())` will group characters using its real value (because `Function.identity()` maps each element into itself) as a **key**, and then will count occurrences of each element  (`counting()`) and use that number as map **value**